[menatahari] iP

menatahari opened the pull request:
# Buto
> "Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes/))
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/menatahari/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

menatahari committed:
Renamed Project, changed greet message

menatahari committed:
Renamed Project, changed greet message

menatahari committed:
echo user command

menatahari committed:
echo user command

menatahari committed:
echo user command

menatahari committed:
echo user command

menatahari committed:
store user command and list stored command

menatahari committed:
store user command and list stored command

menatahari committed:
update line

menatahari committed:
update line

menatahari committed:
Create Task clas

menatahari committed:
Create Task clas

menatahari committed:
Create todo, event, deadline class

menatahari committed:
Create todo, event, deadline class

menatahari committed:
automated text ui testing

menatahari committed:
automated text ui testing

menatahari committed:
Add exception handling

menatahari committed:
Add exception handling

menatahari committed:
Enable deletion of task

menatahari committed:
Enable deletion of task

menatahari committed:
Save task list from previous session

menatahari committed:
Save task list from previous session

menatahari committed:
Merge branch 'Level-7'
"Merge Level-7 no fast-forward"

menatahari committed:
Fix path

menatahari committed:
Fix path

menatahari committed:
Merge branch 'branch-Level-7'
Merge branch-Level-7 no fast forward

menatahari committed:
Parse due date and time for deadline

menatahari committed:
Parse due date and time for deadline

menatahari committed:
Merge branch 'branch-Level-8'
Merge branch-Level-8 no fast forward

menatahari committed:
Create new classes for OOP implementation

menatahari committed:
Create new classes for OOP implementation

menatahari committed:
Merge branch 'branch-A-MoreOOP'
Merge branch-More-OOP no fast forward

menatahari committed:
Add packages

menatahari committed:
Add packages

menatahari committed:
Add gradle support

menatahari committed:
Add JUnit test for Parser and TaskList

menatahari committed:
Add JUnit test for Parser and TaskList

menatahari committed:
Update JUnit test

menatahari committed:
Add javadoc

menatahari committed:
Merge branch-A-Javadoc

menatahari committed:
Apply coding standard

menatahari committed:
Merge branch-A-CodingStandard no fast forward

menatahari committed:
Allow users to find tasks matching a keyword

menatahari committed:
Merge branch-Level-9 no fast forward

tohlh commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,60 @@
+     *
+     * @return String representation of task for data storage.
+     */
+    public String storeData() {
```
Should change to a more descriptive name? Like `toFileString`.

tohlh commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,92 @@
+ */
+public class Storage {
+    private String filePath = "./src/main/";
+    private String directoryPath = "./src/main/";
```
Should we make these constants?

tohlh commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,153 @@
+        printMessage("Nice! I've marked this task as done:");
+        printMessage(TAB + task);
+        printLine();
+    }
```
Should we change this to a more descriptive name? Like `printTaskAsMarked`.

tohlh commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,153 @@
+     *
+     * @param task String representation of unmarked task.
+     */
+    public void printUnmark(String task) {
```
Should we change this to a more descriptive name? Like `printTaskAsUnmarked`.

tohlh reviewed (commented):
Overall, you have done a good job!

However, I would suggest you explore extracting the commands into their separate classes to make the codebase more extensible.

I also notice an excessive use of `setters` and `getters` in classes like `Parser`, `Task`, etc.

It would be nice to explore these. Great job!

Rishit02 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,126 @@
+
+import duke.exception.EmptyTaskNameException;
+import duke.exception.NoTaskTypeException;
+
```
I like that you placed a space in between import statements to make it easier to understand.

Rishit02 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    /**
+     * Find tasks having names matching the given keyword.
```
I like the way that you have created very clear description of your methods.

Rishit02 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,153 @@
+    }
+
+    /**
+     * Fetches the detail and description of the command.
```
Maybe you can replace Fetches with Returns?

Rishit02 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,129 @@
+    }
+
+    /**
+     * Retrieves raw user input.
```
Maybe could replace Retrieves with Returns?

Rishit02 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,92 @@
+ * Handles loading and writing tasks data to a file.
+ */
+public class Storage {
+    private String filePath = "./src/main/";
```
Maybe since these remain constant can make the variable names fully uppercase?

Rishit02 reviewed (commented):
Overall, LGTM. Just a few naming nits to fix.

menatahari committed:
Add simple GUI version of Duke

menatahari committed:
Merge branch-Level-10

menatahari committed:
Add assertions in methods of Response class

For add, mark, unmark, and delete command the task would never be empty.

Assert statements are added to the relevant method to ensure that the task is not empty before processing and confirm no unwanted behaviour happened.

menatahari committed:
Refactor code

Some naming are ambiguous and may confuse the code reader.

Changing the names of the method and constant improve code quality.

As astep towards that let's ensure
* the method complies with coding quality standard
* naming of method and variables are detailed

menatahari committed:
Merge pull request #2 from menatahari/branch-A-Assertions

Add assertion to Response

menatahari committed:
Add C-Tagging extention and refactor code

The new feature allows user to tag their tasks to make it more extensible. Some codes are refactored for better understandability.

Implement the new tagging feature for tasks. Multiple tags can be applied to each task.

In the future:
* Code needs to be refactored further as there are still too long methods.

menatahari committed:
Merge pull request #3 from menatahari/branch-A-CodeQuality

Add C-Tagging extention and refactor code

menatahari committed:
Add resource and code reuse acknowledgement

menatahari committed:
Update README.md

menatahari committed:
Update README.md

menatahari committed:
Update README.md

menatahari committed:
Refactor junit test

menatahari committed:
Change shadowJar build name

menatahari committed:
Fix bug in tagging process and improve gui

menatahari committed:
Add background, update response, and update ui picture

menatahari committed:
Fix typo causing build failure and update xmlns version in fxml file
